### ISSUE TITLE
[Bug]: panic after container [milvus-standalone] start

### ISSUE THREAD

**Issue by kingzcheung** [milvus-io/milvus#16003]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:  v2.0.1
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2): go v2.0
- OS(Ubuntu or CentOS):  Ubuntu
- CPU/Memory: 48 core / 120G mem
- GPU: 2 x GTX 3080
- Others: docker
```

### Current Behavior

I got error after `docker restart milvus-standalone`

```shell
[2022/03/11 01:33:00.580 +00:00] [DEBUG] [service.go:154] ["start Proxy server done"]
[2022/03/11 01:33:00.580 +00:00] [DEBUG] [proxy.go:52] ["Proxy successfully started"]
[2022/03/11 01:33:00.580 +00:00] [DEBUG] [proxy_manager.go:138] ["received proxy put event with session"] [session="{\"ServerID\":16,\"ServerName\":\"proxy\",\"Address\":\"172.19.0.5:19530\",\"TriggerKill\":true}"]
[2022/03/11 01:33:00.580 +00:00] [DEBUG] [timeticksync.go:263] ["Add session for timeticksync"] [serverID=16]
[2022/03/11 01:33:00.580 +00:00] [DEBUG] [session_util.go:388] [WatchService] ["event type"=1]
[2022/03/11 01:33:00.581 +00:00] [DEBUG] [proxy_client_manager.go:88] ["succeed to create proxy client"] [address=172.19.0.5:19530] [serverID=16]
[2022/03/11 01:33:00.698 +00:00] [DEBUG] [segment_loader.go:371] ["Do insert done in segment loader"] [len=415] [segmentID=431727669179318274] [collectionID=431727666387484673]
[2022/03/11 01:33:00.698 +00:00] [DEBUG] [segment_loader.go:219] ["loading bloom filter..."]
[2022/03/11 01:33:00.715 +00:00] [DEBUG] [segment_loader.go:227] ["loading delta..."]
[2022/03/11 01:33:00.715 +00:00] [INFO] [segment_loader.go:479] ["there are no delta logs saved with segment"] [segmentID=431727669179318274]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1ae25ae]

goroutine 485 [running]:
github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).loadSegment(0xc000639e00, 0xc001fa1a98, 0xc000000002, 0x2, 0x2)
	/go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:151 +0x15ae
github.com/milvus-io/milvus/internal/querynode.(*watchDmChannelsTask).Execute(0xc002776f90, 0x34c06d0, 0xc000301fc0, 0x0, 0x0)
	/go/src/github.com/milvus-io/milvus/internal/querynode/task.go:290 +0xf3f
github.com/milvus-io/milvus/internal/querynode.(*taskScheduler).processTask(0xc000a98080, 0x34d35d8, 0xc002776f90, 0x34d1750, 0xc0002760f0)
	/go/src/github.com/milvus-io/milvus/internal/querynode/task_scheduler.go:61 +0x173
github.com/milvus-io/milvus/internal/querynode.(*taskScheduler).taskLoop(0xc000a98080)
	/go/src/github.com/milvus-io/milvus/internal/querynode/task_scheduler.go:78 +0x15a
created by github.com/milvus-io/milvus/internal/querynode.(*taskScheduler).Start
	/go/src/github.com/milvus-io/milvus/internal/querynode/task_scheduler.go:86 +0x5c
```

### Expected Behavior

Service started

**Comment by yanliang567** [comment]
@zhuwenxing could you please help to reproduce it locally?
/assign @zhuwenxing 
/unassign

**Comment by zhuwenxing** [comment]
it looks like the issue https://github.com/milvus-io/milvus/issues/15713
I will check this again!

**Comment by kingzcheung** [comment]
> it looks like the issue #15713 I will check this again!

its ok on milvus-dev:v2.0.2

**Comment by xiaofan-luan** [comment]
> > it looks like the issue #15713 I will check this again!
> 
> It's ok on milvus-dev:v2.0.2.But i fail to search:
> 
> ```
> [2022/03/11 03:29:23.248 +00:00] [ERROR] [task_scheduler.go:501] ["Failed to post-execute task: searchTask:wait to finish failed, timeout: 431743126631612420"] [traceID=3a0260e4ead4327f] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:501"]
> [2022/03/11 03:29:23.248 +00:00] [WARN] [impl.go:2489] ["Search failed to WaitToFinish"] [error="proxy TaskCondition context Done"] [traceID=3a0260e4ead4327f] [role=proxy] [msgID=431743126631612420] [db=] [collection=test] [partitions="[]"] [dsl=] [len(PlaceholderGroup)=8204] [OutputFields="[id,category]"] [search_params="[{\"key\":\"round_decimal\",\"value\":\"-1\"},{\"key\":\"anns_field\",\"value\":\"embedding\"},{\"key\":\"topk\",\"value\":\"5\"},{\"key\":\"params\",\"value\":\"{\\\"nprobe\\\":10}\"},{\"key\":\"metric_type\",\"value\":\"L2\"}]"] [travel_timestamp=0] [guarantee_timestamp=0]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]parsed scheme: \"\""]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]scheme \"\" not registered, fallback to default scheme"]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]ccResolverWrapper: sending update to cc: {[{172.18.0.5:53100  <nil> 0 <nil>}] <nil> <nil>}"]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]ClientConn switching balancer to \"pick_first\""]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]Channel switches to new LB policy \"pick_first\""]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to CONNECTING"]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]Subchannel picks a new address \"172.18.0.5:53100\" to connect"]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc003123e80, {CONNECTING <nil>}"]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to CONNECTING"]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to READY"]
> [2022/03/11 03:29:35.149 +00:00] [INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc003123e80, {READY <nil>}"]
> [2022/03/11 03:29:35.149 +00:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to READY"]
> [2022/03/11 03:29:42.344 +00:00] [WARN] [impl.go:2489] ["Search failed to WaitToFinish"] [error="proxy TaskCondition context Done"] [traceID=4d06de456b179b98] [role=proxy] [msgID=431743126631612438] [db=] [collection=test] [partitions="[]"] [dsl=] [len(PlaceholderGroup)=8204] [OutputFields="[id,category]"] [search_params="[{\"key\":\"params\",\"value\":\"{\\\"nprobe\\\":10}\"},{\"key\":\"metric_type\",\"value\":\"L2\"},{\"key\":\"round_decimal\",\"value\":\"-1\"},{\"key\":\"anns_field\",\"value\":\"embedding\"},{\"key\":\"topk\",\"value\":\"5\"}]"] [travel_timestamp=0] [guarantee_timestamp=0]
> [2022/03/11 03:29:42.344 +00:00] [ERROR] [task_scheduler.go:501] ["Failed to post-execute task: searchTask:wait to finish failed, timeout: 431743126631612438"] [traceID=4d06de456b179b98] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:501"]
> [2022/03/11 03:29:44.204 +00:00] [WARN] [query_coord.go:532] ["loadBalanceSegmentLoop: there are too few available query nodes to balance"] [onlineNodeIDs="[34]"] [availableNodeIDs="[34]"]
> ```

could you offer the nodes of querynodes and querycoord?

**Comment by zhuwenxing** [comment]
> > it looks like the issue #15713 I will check this again!
> 
> It's ok on milvus-dev:v2.0.2.But i fail to search:
> 
> ```
> [2022/03/11 03:29:23.248 +00:00] [ERROR] [task_scheduler.go:501] ["Failed to post-execute task: searchTask:wait to finish failed, timeout: 431743126631612420"] [traceID=3a0260e4ead4327f] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:501"]
> [2022/03/11 03:29:23.248 +00:00] [WARN] [impl.go:2489] ["Search failed to WaitToFinish"] [error="proxy TaskCondition context Done"] [traceID=3a0260e4ead4327f] [role=proxy] [msgID=431743126631612420] [db=] [collection=test] [partitions="[]"] [dsl=] [len(PlaceholderGroup)=8204] [OutputFields="[id,category]"] [search_params="[{\"key\":\"round_decimal\",\"value\":\"-1\"},{\"key\":\"anns_field\",\"value\":\"embedding\"},{\"key\":\"topk\",\"value\":\"5\"},{\"key\":\"params\",\"value\":\"{\\\"nprobe\\\":10}\"},{\"key\":\"metric_type\",\"value\":\"L2\"}]"] [travel_timestamp=0] [guarantee_timestamp=0]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]parsed scheme: \"\""]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]scheme \"\" not registered, fallback to default scheme"]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]ccResolverWrapper: sending update to cc: {[{172.18.0.5:53100  <nil> 0 <nil>}] <nil> <nil>}"]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]ClientConn switching balancer to \"pick_first\""]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]Channel switches to new LB policy \"pick_first\""]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to CONNECTING"]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]Subchannel picks a new address \"172.18.0.5:53100\" to connect"]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc003123e80, {CONNECTING <nil>}"]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to CONNECTING"]
> [2022/03/11 03:29:35.148 +00:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to READY"]
> [2022/03/11 03:29:35.149 +00:00] [INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc003123e80, {READY <nil>}"]
> [2022/03/11 03:29:35.149 +00:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to READY"]
> [2022/03/11 03:29:42.344 +00:00] [WARN] [impl.go:2489] ["Search failed to WaitToFinish"] [error="proxy TaskCondition context Done"] [traceID=4d06de456b179b98] [role=proxy] [msgID=431743126631612438] [db=] [collection=test] [partitions="[]"] [dsl=] [len(PlaceholderGroup)=8204] [OutputFields="[id,category]"] [search_params="[{\"key\":\"params\",\"value\":\"{\\\"nprobe\\\":10}\"},{\"key\":\"metric_type\",\"value\":\"L2\"},{\"key\":\"round_decimal\",\"value\":\"-1\"},{\"key\":\"anns_field\",\"value\":\"embedding\"},{\"key\":\"topk\",\"value\":\"5\"}]"] [travel_timestamp=0] [guarantee_timestamp=0]
> [2022/03/11 03:29:42.344 +00:00] [ERROR] [task_scheduler.go:501] ["Failed to post-execute task: searchTask:wait to finish failed, timeout: 431743126631612438"] [traceID=4d06de456b179b98] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:501"]
> [2022/03/11 03:29:44.204 +00:00] [WARN] [query_coord.go:532] ["loadBalanceSegmentLoop: there are too few available query nodes to balance"] [onlineNodeIDs="[34]"] [availableNodeIDs="[34]"]
> ```

@kingzcheung 
I tried to reproduce this with same steps in issue https://github.com/milvus-io/milvus/issues/15713, but it was not reproduced.

Can you open another issue for this `fail to search` and please attach the log?
/assign @kingzcheung

**Comment by kingzcheung** [comment]
Sorry, I went to do something else. 

I forgot the steps about going from import to search.

**Comment by xiaofan-luan** [comment]
similar to #16147

**Comment by xiaofan-luan** [comment]
duplicated and should be fixed on 2.0.2
